fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -1071,7 +1082,12 @@ class ApplyMetadataMutationsImpl {
 
 				auto startItr = tenantMap->lower_bound(startTenant);
 				auto endItr = tenantMap->lower_bound(endTenant);
+
+				auto startIdItr = tenantIdNameMap->lower_bound(startItr->second.id);
+				auto endIdItr = tenantIdNameMap->lower_bound(endItr->second.id);
```

Comment:
I don't think this will work since the IDs are not ordered in the same way. You will have to iterate over all removed tenants and remove them from the index one by one.